### PR TITLE
EN-24494: cannot merge chained soql if there is search and grouping a…

### PR DESCRIPTION
**Pull request by chitang**
…ny more after the new semantics of search (application to the output schema instead of the input schema)

### CODE REVIEW

File: version.sbt [hunk 1]
```diff
@@ -1 +1 @@
-version in ThisBuild := "2.16.0"
+version in ThisBuild := "2.16.1-SNAPSHOT"
```
Comment (langlk) [inline version.sbt:1]: Is this supposed to say `SNAPSHOT` or should is just by `2.16.1`?

File: soql-analyzer/src/main/scala/com/socrata/soql/SoQLAnalyzer.scala [hunk 2]
```diff
@@ -575,7 +575,7 @@ private class Merger[T](andFunction: MonomorphicFunction[T]) {
   // for a query that contains a search.
   private def tryMerge(a: Analysis, b: Analysis): Option[Analysis] = (a, b) match {
     case (a, _) if (hasWindowFunction(a)) => None
-    case (SoQLAnalysis(aIsGroup, false, aSelect, Nil, aWhere, aGroup, aHaving, aOrder, aLim, aOff, aSearch),
+    case (SoQLAnalysis(aIsGroup, false, aSelect, Nil, aWhere, aGroup, aHaving, aOrder, aLim, aOff, None),
```
Comment (langlk) [inline soql-analyzer/src/main/scala/com/socrata/soql/SoQLAnalyzer.scala:578]: Are we adding the search to the query elsewhere? It looks like all of these just remove it, I expect I'm missing something though.

File: version.sbt (hunk 1 above)
Comment (chitang) [reply version.sbt:1]: snapshot now and remove snapshot in two steps is proper.  We skip a lot these days.

File: soql-analyzer/src/main/scala/com/socrata/soql/SoQLAnalyzer.scala (hunk 2 above)
Comment (chitang) [reply soql-analyzer/src/main/scala/com/socrata/soql/SoQLAnalyzer.scala:578]: These search cases just won't match any more and go to the catch all case which means not mergeable.  Then the chained soql retains it's original form with search.

Comment (chitang) [reply soql-analyzer/src/main/scala/com/socrata/soql/SoQLAnalyzer.scala:578]: These search cases just won't match any more and go to the catch all case which means not mergeable.  Then the chained soql retains it's original form with search.

**Review by langlk — APPROVED** [review]

**Pull request merged**